ICSSoft.STORMNET.Business/SQLDataService/SQLDataService.cs

```diff
@@ -4963,9 +4769,15 @@ private void GetOrderFromDependenciesHelper(Type currentType, Dictionary<Type, L
 
                             foreach (STORMDO.View subview in views)
                             {
-                                DataObject[] detailsObjects;
-                                IEnumerable<DataObject> extraProcessingObjects =
-                                    AddDeletedViewToDeleteDictionary(subview, deleteDictionary, processingObject.__PrimaryKey, out detailsObjects, deleteTables, tableOperations, dataObjectCache, dbTransactionWrapper);
+                                var result = dbTransactionWrapper switch
+                                {
+                                    DbTransactionWrapper syncWrapper => AddDeletedViewToDeleteDictionary(subview, deleteDictionary, processingObject.__PrimaryKey, deleteTables, tableOperations, dataObjectCache, syncWrapper),
+                                    DbTransactionWrapperAsync asyncWrapper => Task.Run(() => AddDeletedViewToDeleteDictionaryAsync(subview, deleteDictionary, processingObject.__PrimaryKey, deleteTables, tableOperations, dataObjectCache, asyncWrapper)).GetAwaiter().GetResult(),
```

Comment:
Это без сомнений костыль, но благо в эту ветку не так часто ходит, да и сам аудит синхронный
```suggestion
                                    DbTransactionWrapperAsync asyncWrapper => AddDeletedViewToDeleteDictionaryAsync(subview, deleteDictionary, processingObject.__PrimaryKey, deleteTables, tableOperations, dataObjectCache, asyncWrapper).GetAwaiter().GetResult(),
```

Reply:
Ок